app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java

```diff
@@ -1,49 +1,164 @@
+import android.util.Log;
 import android.view.View;
+import android.widget.ImageView;
 import android.widget.PopupMenu;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.Navigation;
 import androidx.recyclerview.widget.RecyclerView;
 
+import com.bumptech.glide.Glide;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.Function;
+
 import ch.epfl.favo.R;
+import ch.epfl.favo.exception.IllegalAcceptException;
+import ch.epfl.favo.exception.IllegalRequestException;
 import ch.epfl.favo.favor.Favor;
+import ch.epfl.favo.user.UserUtil;
+import ch.epfl.favo.util.CommonTools;
+import ch.epfl.favo.util.DependencyFactory;
+import ch.epfl.favo.viewmodel.IFavorViewModel;
+
+import static ch.epfl.favo.util.CommonTools.getSnackbarMessageForFailedRequest;
+import static ch.epfl.favo.util.CommonTools.handleException;
 
 class FavorViewHolder extends RecyclerView.ViewHolder {
 
+  private String TAG = "FavorViewHolder";
+
   private TextView mTitleView;
-  private TextView mDescriptionView;
+  private TextView mRequesterView;
+  private TextView mRewardView;
+  private ImageView mRequesterIconView;
   private TextView buttonViewOption;
+  private IFavorViewModel favorViewModel;
+
+  private PopupMenu popup;
+
+  private FirebaseUser currentUser;
+  private Resources resources;
 
   FavorViewHolder(@NonNull View itemView) {
     super(itemView);
 
     mTitleView = itemView.findViewById(R.id.item_title);
-    mDescriptionView = itemView.findViewById(R.id.item_desc);
     buttonViewOption = itemView.findViewById(R.id.item_menu_btn);
+    mRequesterView = itemView.findViewById(R.id.item_requester);
+    mRequesterIconView = itemView.findViewById(R.id.item_icon);
+    mRewardView = itemView.findViewById(R.id.item_coins);
+
+    currentUser = DependencyFactory.getCurrentFirebaseUser();
   }
 
-  void bind(Context context, @NonNull Favor favor) {
+  @RequiresApi(api = Build.VERSION_CODES.N)
+  void bind(Context context, @NonNull Favor favor, View parentView) {
+    resources = context.getResources();
+    favorViewModel = (IFavorViewModel) new ViewModelProvider((FragmentActivity) context)
+            .get(DependencyFactory.getCurrentViewModelClass());
+
+    setItemDisplay(favor, context);
+
+    buttonViewOption.setOnClickListener(view -> {
+      popup = new PopupMenu(context, buttonViewOption);
+      popup.inflate(R.menu.item_menu_active);
+
+      // display "commit" option if the favor is requested by the current user.
+      boolean isRequestedByCurrentUser = currentUser.getUid().equals(favor.getRequesterId());
+      if (isRequestedByCurrentUser)
+        popup.getMenu().findItem(R.id.item_menu_commit).setEnabled(false);
```

Comment:
Personally, I thought it is better to set it invisible.

Reply:
Also, it is not always reasonable to show "cancel" MenuItem for users, especially when it is posted by others or already archived.

Reply:
I agree, I would remove the commit. For the cancel, this should be shown only for favors created by the current user logged in.